File: spepjava/src/com/qut/middleware/spep/authn/AuthnProcessorData.java
/* 
 * Copyright 2006, Queensland University of Technology
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not 
 * use this file except in compliance with the License. You may obtain a copy of 
 * the License at 
 * 
 *   http://www.apache.org/licenses/LICENSE-2.0 
 * 
 * Unless required by applicable law or agreed to in writing, software 
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT 
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the 
 * License for the specific language governing permissions and limitations under 
 * the License.
 * 
 * Author: Shaun Mangelsdorf
 * Creation Date: 24/11/2006
 * 
 * Purpose: Interface for storing intermediate Authentication data while the session
 * 		is being established.
 */
package com.qut.middleware.spep.authn;



/** Interface for storing intermediate Authentication data while the session
 * 		is being established.*/
public interface AuthnProcessorData
{
	/** Get the URL to be used to send Authn Requests to.
	 * 
	 * @return The request URL
	 */
	public String getRequestURL();
	
	/** Set the URL to be used to send Authn Requests to.
	 * 
	 * @param requestURL The request URL
	 */
	public void setRequestURL(String requestURL);
	
	/** Get the client session ID used in authenticated sessions.
	 * 
	 * @return The session ID.
	 */
	public String getSessionID();
	
	/** Set the client session ID used in authenticated sessions.
	 * 
	 * @param sessionID The session ID
	 */
	public void setSessionID(String sessionID);
	
	public String getBindingIdentifier();
	
	public void setBindingIdentifier(String bindingIdentifier);
	
	public boolean isReturningRequest();
	
	public void setReturningRequest();
	
	public String getDestinationURL();
	
	public void setDestinationURL(String destinationURL);

	public String getSSORequestServerName();
	
	public void setSSORequestServerName(String ssoRequestServerName);
	
	public String getSSORequestURI();
	
	public void setSSORequestURI(String ssoRequestURI);
}

File: spepcpp/src/spep/authn/AuthnProcessorData.cpp
/* Copyright 2006-2007, Queensland University of Technology
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not 
 * use this file except in compliance with the License. You may obtain a copy of 
 * the License at 
 * 
 *   http://www.apache.org/licenses/LICENSE-2.0 
 * 
 * Unless required by applicable law or agreed to in writing, software 
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT 
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the 
 * License for the specific language governing permissions and limitations under 
 * the License.
 * 
 * Author: Shaun Mangelsdorf
 * Creation Date: 19/02/2007
 * 
 * Purpose: 
 */

#include "spep/authn/AuthnProcessorData.h"

namespace spep {

AuthnProcessorData::AuthnProcessorData()
{
	_disableAttributeQuery = false;
}

std::string AuthnProcessorData::getRequestURL() const
{
	return _requestURL;
}

void AuthnProcessorData::setRequestURL(const std::string &requestURL)
{
	_requestURL = requestURL;
}

std::string AuthnProcessorData::getBaseRequestURL() const
{
	return _baseRequestURL;
}

void AuthnProcessorData::setBaseRequestURL(const std::string& baseRequestURL)
{
	_baseRequestURL = baseRequestURL;
}

std::string AuthnProcessorData::getSessionID() const
{
	return _sessionID;
}

void AuthnProcessorData::setSessionID(const std::string &sessionID)
{
	_sessionID = sessionID;
}

const saml2::SAMLDocument& AuthnProcessorData::getRequestDocument() const
{
	return _requestDocument;
}

void AuthnProcessorData::setRequestDocument(const saml2::SAMLDocument& document)
{
	_requestDocument = document;
}

const saml2::SAMLDocument& AuthnProcessorData::getResponseDocument() const
{
	return _responseDocument;
}

void AuthnProcessorData::setResponseDocument(const saml2::SAMLDocument& document)
{
	_responseDocument = document;
}

void AuthnProcessorData::setDisableAttributeQuery(bool value)
{
	_disableAttributeQuery = value;
}

bool AuthnProcessorData::getDisableAttributeQuery() const
{
	return _disableAttributeQuery;
}

void AuthnProcessorData::setRemoteIpAddress(const std::string &remoteIpAddress)
{
	_remoteIpAddress = remoteIpAddress;
}

std::string AuthnProcessorData::getRemoteIpAddress() const
{
	return _remoteIpAddress;
}

}